### PR TITLE
[202311] Fix pstore service not running

### PR DESCRIPTION
**Pull request by DavidZagury**
#### Why I did it
The pstore service is not able to start, even if there are backtrace for kernel panics/oops in the UEFI.
This carry a risk of filling up the UEFI storage space.
This fixes the issue - https://github.com/sonic-net/sonic-buildimage/issues/18895

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Systemd on newer version has fixed this issue by adding a dependency on the kernel module to ensure we only scan for
pstate after we can actually see pstate.
Took this fix to our system.

#### How to verify it
Simulate a kernel crash, and make sure that when system is live again, the backtrace are located on /var/lib/systemd/pstore/ and not /sys/fs/pstore/

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by dprital — APPROVED** [review]

**Comment by dprital** [comment]
@saiarcot895 , Can you please approve this PR ?

File: build_debian.sh [hunk 1]
```diff
@@ -526,6 +526,9 @@ rm /files/lib/systemd/system/rsyslog.service/Service/ExecStart/arguments
 set /files/lib/systemd/system/rsyslog.service/Service/ExecStart/arguments/1 -n
 "
 
+# Ensure pstore service is running
+sudo cp files/image_config/system-pstore/systemd-pstore.service $FILESYSTEM_ROOT/lib/systemd/system/systemd-pstore.service
```
[the remark this answers is not in the document]
Comment (DavidZagury) [reply build_debian.sh]: Updated

**Comment by dprital** [comment]
@saiarcot895 , Can you please review and approve ?

**Review by saiarcot895 — APPROVED** [review]

**Comment by yxieca** [comment]
@DavidZagury @dprital can you confirm that this change is only needed for 202311? master branch has fix by default already? I think your PR comments implies this but please confirm.

**Comment by DavidZagury** [comment]
@yxieca, yes, this PR fixes a bug in Systemd version 247.3 (https://github.com/systemd/systemd/issues/18540)

This bug was fixed on newer versions of Systemd, Debian Bookworm already uses the version with the fix.
Systemd version 247.3 is used on Debian Bullseye, so this PR brings the same correct behavior to 202311.

It can be used in other branches based on Bullseye, but since bookwork (and 202405) has the fix already it is not needed for them.

**Pull request merged**